src/extension.ts

```diff
@@ -109,36 +160,19 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
 
       kubeconfig.exportToFile(config, kubeconfigFile);
 
-      // check if cluster is accessible
-      const status = await getConnectionStatus(apiURL, token);
-      
-      const disposable = provider.registerKubernetesProviderConnection({
-        name: params[ContextNameParam], 
-        status: () => status, 
-        endpoint: {
-          apiURL
-        }, lifecycle: {
-          delete: async () => {
-            return;
-          }
-        }
-      });
-      registeredConnections.push(disposable);
+      await registerConnection(params[ContextNameParam], apiURL, token);
     },
     creationDisplayName: ProvideDisplayName,
   });
 
   extensionContext.subscriptions.push(provider);
-  // run update connections once to load existing connections to avoid 2s delay
-  updateConnections().then (() => {
-    updateConnectionsPreiodically()
-  });
+  updateConnectionsPreiodically();
 }
 
 function updateConnectionsPreiodically(): void {
-   updateConnectionTimeout = setTimeout(() => {
-    updateConnections().then(updateConnectionsPreiodically);
-   }, 2000);
+   updateConnections().then(() => {
+    updateConnectionTimeout = setTimeout(updateConnectionsPreiodically,2000);
```

Comment:
small typo in the name of the function
```suggestion
    updateConnectionTimeout = setTimeout(updateConnectionsPeriodically,2000);
```

Reply:
sorry, installing spell check extension.